symengine/mul.cpp

```diff
@@ -63,7 +63,7 @@ bool Mul::is_canonical(const RCP<const Number> &coef,
         // e.g. x**2**y (={x**2:y}), which should be represented as x**(2y)
         //     (={x:2y})
         if (is_a<Pow>(*p.first))
-            return false;
+            return true;
```

Comment:
Then there must be another error. Since you are returning prematurely here another error might not have shown up.